transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -46,18 +47,26 @@
     //    other value T    when EL is waiting with wakeup scheduled at time T
     private final AtomicLong nextWakeupNanos = new AtomicLong(AWAKE);
     private final FileDescriptor eventfd;
-    private final IovecArrayPool iovecArrayPool;
+
+    private final IovArray[] iovArrays;
+    private int iovArrayIdx;
 
     private long prevDeadlineNanos = NONE;
     private boolean pendingWakeup;
 
     IOUringEventLoop(final EventLoopGroup parent, final Executor executor, final boolean addTaskWakesUp) {
         super(parent, executor, addTaskWakesUp);
+        // Ensure that we load all native bits as otherwise it may fail when try to use native methods in IovArray
+        IOUring.ensureAvailability();
 
+        // Let's hard code this to 8 IovArrays to keep the memory overhead kind of small.
+        iovArrays = new IovArray[8];
```

Comment:
we could even make this more optimal by writing a custom `IovArray` implementation. But for now this seems to work well enough and allows us for code-reuse.

Reply:
I like that, 8 we should take it as default value when it's configurable